src/main/java/lotto/domain/LottoRank.java

```diff
@@ -1,41 +1,44 @@
 package lotto.domain;
 
 public enum LottoRank {
-    ZERO(0, 0L),
-    ONE(1, 0L),
-    TWO(2, 0L),
-    THREE(3, 5_000L),
-    FOUR(4, 50_000L),
-    FIVE(5, 1_500_000L),
-    SIX(6, 2_000_000_000L);
-
-    private final int number;
-    private final Long prizeMoney;
-
-    LottoRank(int number, Long prizeMoney) {
-        this.number = number;
+    MISS(0, 0L),
+    FIFTH(3, 5_000L),
+    FOURTH(4, 50_000L),
+    THIRD(5, 1_500_000L),
+    SECOND(5, 30_000_000L),
+    FIRST(6, 2_000_000_000L);
+
+    private final int count;
+    private final long prizeMoney;
+
+    LottoRank(int count, long prizeMoney) {
+        this.count = count;
         this.prizeMoney = prizeMoney;
     }
 
-    public int matchNumber() {
-        return this.number;
-    }
-
-    public Long prize() {
-        return this.prizeMoney;
-    }
+    public static LottoRank findMatchCount(int matchNumber, boolean bouns) {
+        if (matchNumber == 5 && bouns) {
+            return LottoRank.SECOND;
+        }
 
-    public static LottoRank findMatchNumber(int matchNumber) {
         for (LottoRank lottoRank : LottoRank.values()) {
-            if (lottoRank.number == matchNumber) {
+            if (lottoRank.count == matchNumber) {
```

Comment:
```suggestion
            if (lottoRank.sameNumber(matchNumber)) {
                return lottoRank;
            }
            
            ...
            
                private boolean sameNumber(int matchNumber) {
        return number == matchNumber;
    }

```
객체에게 비교하도록 메소드로 분리해볼수도 있겠네요~